src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProjectProvider.cs

```diff
@@ -19,15 +20,27 @@ public class LegacyCSProjPackageReferenceProjectProvider : IProjectSystemProvide
     {
         private readonly IProjectSystemCache _projectSystemCache;
 
+        private readonly Lazy<IComponentModel> _componentModel;
+
         [ImportingConstructor]
-        public LegacyCSProjPackageReferenceProjectProvider(IProjectSystemCache projectSystemCache)
+        public LegacyCSProjPackageReferenceProjectProvider(
+            IProjectSystemCache projectSystemCache,
+            [Import(typeof(SVsServiceProvider))]
+            IServiceProvider serviceProvider)
```

Comment:
Underlying code in `NuGetUIThreadHelper` consumes the only interface `IProjectServiceAccessor` which can be directly MEF imported here.